hack/build/docker/builder/Dockerfile

```diff
@@ -33,7 +33,7 @@ RUN pip3 install --upgrade j2cli operator-courier==2.1.11 && \
 	ln -s /opt/gradle/gradle-6.6/bin/gradle /usr/local/bin/gradle && \
 	rm gradle-6.6-bin.zip
 
-ENV GIMME_GO_VERSION=1.17.10 GOPATH="/go" GO111MODULE="on"
+ENV GIMME_GO_VERSION=1.18.3 GOPATH="/go" GO111MODULE="on"
```

Comment:
Have you confirmed that no other changes will be required to the builder image in order to compile with new compiler?  Just want to make sure to reduce churn

Reply:
Like any bazel changes required?

Reply:
I think [this](https://github.com/maya-r/containerized-data-importer/commit/b0d5b50b1f3ebf4e2b29149f879647a882dab530) was the change I did when I tried it out

Reply:
I really didn't want to get into a new bazel version, but I suppose I can add that. Can only do the builder in this PR, then the vendoring and other stuff in a follow up that uses the builder.

Reply:
Well this part is just for the builder, the actual code will use the builder and then update libraries/etc.

Reply:
yeah updating io_bazel_rules_go seems to make it work